lib/measures/tbd/measure.rb

```diff
@@ -53,122 +53,296 @@ def modeler_description
-    alter_model.setDefaultValue(true)
-    args << alter_model
-
-    write_tbd_json = OpenStudio::Measure::OSArgument.makeBoolArgument("write_tbd_json", true, false)
-    write_tbd_json.setDisplayName("Write 'tbd.out.json'")
-    write_tbd_json.setDescription("Write out 'tbd.out.json' file e.g., to customize for subsequent runs (edit, and place under '/files' as 'tbd.json').")
-    write_tbd_json.setDefaultValue(false)
-    args << write_tbd_json
-
-    gen_UA_report = OpenStudio::Measure::OSArgument.makeBoolArgument("gen_UA_report", true, false)
-    gen_UA_report.setDisplayName("Generate UA' report")
-    gen_UA_report.setDescription("Compare ∑U•A + ∑PSI•L + ∑KHI•n : 'Design' vs UA' reference (see pull-down option below).")
-    gen_UA_report.setDefaultValue(false)
-    args << gen_UA_report
-
-    ua_reference = OpenStudio::Measure::OSArgument.makeChoiceArgument("ua_reference", choices, true)
-    ua_reference.setDisplayName("UA' reference")
-    ua_reference.setDescription("e.g. 'poor', 'regular', 'efficient', 'code'.")
-    ua_reference.setDefaultValue("code (Quebec)")
-    args << ua_reference
-
-    gen_kiva = OpenStudio::Measure::OSArgument.makeBoolArgument("gen_kiva", true, false)
+    arg = "write_tbd_json"
+    dsc = "Write out 'tbd.out.json' file e.g., to customize for subsequent "   \
+          "runs (edit, and place under '/files' as 'tbd.json')."
+    write_tbd = OpenStudio::Measure::OSArgument.makeBoolArgument(arg, false)
+    write_tbd.setDisplayName("Write 'tbd.out.json'")
+    write_tbd.setDescription(dsc)
+    write_tbd.setDefaultValue(false)
+    args << write_tbd
+
+    walls = {c: {}, dft: "ALL wall constructions"}
+    roofs = {c: {}, dft: "ALL roof constructions"}
+    flors = {c: {}, dft: "ALL floor constructions"}
+    walls[:c][walls[:dft]] = {a: 100000000000000}
+    roofs[:c][roofs[:dft]] = {a: 100000000000000}
+    flors[:c][flors[:dft]] = {a: 100000000000000}
+    walls[:chx] = OpenStudio::StringVector.new
+    roofs[:chx] = OpenStudio::StringVector.new
+    flors[:chx] = OpenStudio::StringVector.new
+
+    if model
+      model.getSurfaces.each do |s|
+        type = s.surfaceType.downcase
+        next unless type == "wall" || type == "roofceiling" || type == "floor"
+        next unless s.outsideBoundaryCondition.downcase == "outdoors"
+        next if s.construction.empty?
+        next if s.construction.get.to_LayeredConstruction.empty?
+        lc = s.construction.get.to_LayeredConstruction.get
+        next if walls[:c].key?(lc.nameString)
+        next if roofs[:c].key?(lc.nameString)
+        next if flors[:c].key?(lc.nameString)
+        walls[:c][lc.nameString] = {a: lc.getNetArea} if type == "wall"
+        roofs[:c][lc.nameString] = {a: lc.getNetArea} if type == "roofceiling"
+        flors[:c][lc.nameString] = {a: lc.getNetArea} if type == "floor"
+      end
+
+      walls[:c] = walls[:c].sort_by{ |k,v| v[:a] }.reverse!.to_h
```

Comment:
I think you are sorting constructions by area here?  That seems like a nice touch!

Reply:
And the default should be the first option?

Reply:
See Figure with "NONE" option:

1. "ALL roof constructions" (all-encompassing option - likely the most common option to be picked)
2. roof constructions in sequence, in order of prevalence (largest area first, smallest area last)
3. "NONE" as last (default) item